### PR TITLE
Work around build problem with protos on non-Linux systems

### PR DESCRIPTION
**Pull request by jensck**
This is a workaround for the problem described here: https://github.com/particle-iot/firmware-protobuf/issues/8

*tl;dr:* there are two messages in `common.proto`: `IpAddress` and `IPAddress`.  This all works fine on case-sensitive systems like Linux, but on Windows & Mac it causes one of the files to clobber the other.

The current workaround is to remove the deprecated `IPAddress` message from the proto file in my local repo before generating the Java source.  The real fix will come from the issue linked to above.

@RaimundasSakalauskas you can skip the 2nd and 3rd commits, they didn't end up fixing the problem.  I would have just removed them from git history, but that didn't seem worth the risk if I screwed something up.

### CODE REVIEW

**Comment by jensck** [comment]
@RaimundasSakalauskas look, I made a small PR, too! (Small if you don't include the generated source files anyway...)

**Comment by RaimundasSakalauskas** [comment]
So many commits and the only 1 that matters shows up as summary as it's too long to display.. anyways - looking good and I remember that IP vs Ip thing caused me some confusion too.

**Comment by jensck** [comment]
> So many commits and the only 1 that matters shows up as summary as it's too long to display.. anyways - looking good and I remember that IP vs Ip thing caused me some confusion too.

Yeah, the commit message was the real content there.  Anyway, thanks for the review :+1:

**Comment by jensck** [comment]
Merged, closing.
